### PR TITLE
esdt-integration-tests

### PR DESCRIPTION
**Pull request by sasurobert**
Return value asynccallback on error
Transfer and execute for ESDT
New arwen release

### CODE REVIEW

File: integrationTests/vm/esdt/esdtRoles_test.go [hunk 1]
```diff
@@ -281,3 +283,128 @@ func unsetRole(nodes []*integrationTests.TestProcessorNode, addrForRole []byte,
 		"@" + hex.EncodeToString(roles)
 	integrationTests.CreateAndSendTransaction(tokenIssuer, nodes, big.NewInt(0), vm.ESDTSCAddress, txData, core.MinMetaTxExtraGasCost)
 }
+
+func TestESDTMintAndTransferAndExecute(t *testing.T) {
```
Comment (bogdan-rosianu) [inline integrationTests/vm/esdt/esdtRoles_test.go:287]: `TestESDTMintTransferAndExecute`

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module github.com/ElrondNetwork/elrond-go
 go 1.13
 
 require (
-	github.com/ElrondNetwork/arwen-wasm-vm v1.2.2
+	github.com/ElrondNetwork/arwen-wasm-vm v1.2.3-0.20210408170208-68205f586721
```
Comment (iulianpascalau) [inline go.mod:6]: can we use a proper release tag?

File: process/factory/metachain/vmContainerFactory.go [hunk 3]
```diff
@@ -241,6 +244,11 @@ func (vmf *vmContainerFactory) SystemSmartContractContainer() vm.SystemSCContain
 	return vmf.systemContracts
 }
 
+// SystemSmartContractContainerFactory return the system smart contract container factory
```
Comment (bogdan-rosianu) [inline process/factory/metachain/vmContainerFactory.go:247]: returns*

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (sasurobert) [reply go.mod:6]: after review on arwen yes :)

File: integrationTests/vm/esdt/esdtRoles_test.go (hunk 1 above)
Comment (sasurobert) [reply integrationTests/vm/esdt/esdtRoles_test.go:287]: done

File: process/factory/metachain/vmContainerFactory.go (hunk 3 above)
Comment (sasurobert) [reply process/factory/metachain/vmContainerFactory.go:247]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**